Storage/SQL/SQLiteBookmarks.swift

```diff
@@ -166,30 +166,21 @@ public class SQLiteBookmarks: BookmarksModelFactory {
     }
 
     public func clearBookmarks() -> Success {
-        return self.db.run("DELETE FROM \(TableBookmarks) WHERE parent IS NOT \(BookmarkRoots.RootID)")
+        return self.db.run([
+            ("DELETE FROM \(TableBookmarks) WHERE parent IS NOT ?", [BookmarkRoots.RootID]),
+            self.favicons.getCleanupCommands()
+        ])
     }
 
-    private func runSQL(sql: String, args: Args?, success: (Bool) -> Void, failure: (Any) -> Void) {
-        var err: NSError?
-        self.db.withWritableConnection(&err) { (connection: SQLiteDBConnection, inout err: NSError?) -> Int in
-            if let err = connection.executeChange(sql, withArgs: args) {
-                failure(err)
-                return 0
-            }
-            success(true)
-            return 1
-        }
-    }
-
-    public func removeByURL(url: String, success: (Bool) -> Void, failure: (Any) -> Void) {
+    public func removeByURL(url: String) -> Success {
         log.debug("Removing bookmark \(url).")
-        let sql = "DELETE FROM \(TableBookmarks) WHERE url = ?"
-        let args: Args = [url]
-
-        self.runSQL(sql, args: args, success: success, failure: failure)
+        return self.db.run([
+            ("DELETE FROM \(TableBookmarks) WHERE url = ?", [url]),
+            self.favicons.getCleanupCommands()
```

Comment:
I'm pretty sure we don't want to run a query this heavy on each bookmark unstar, and in any case it's unlikely to help — most bookmarks with favicons will also be visited history, which will retain the icon.

Can we just rely on hooking into history cleanup to do the work?